packages/taco/src/taco-config-validator.ts

```diff
@@ -0,0 +1,339 @@
+ * Use these constants instead of hardcoded strings for better maintainability.
+ */
+export const DOMAIN_NAMES = {
+  /** DEVNET domain ('lynx') - Bleeding-edge developer network */
+  DEVNET: 'lynx',
+  /** TESTNET domain ('tapir') - Stable testnet for current TACo release */
+  TESTNET: 'tapir',
+  /** MAINNET domain ('mainnet') - Production network */
+  MAINNET: 'mainnet',
+} as const;
+
+/**
+ * Domain-ChainId validation type - creates a discriminated union
+ * where each domain is linked to its specific chainId
+ */
+export type DomainChainConfig = {
+  [K in keyof typeof DOMAINS]: {
+    domain: (typeof DOMAINS)[K]['domain'];
+    chainId: (typeof DOMAINS)[K]['chainId'];
+  };
+}[keyof typeof DOMAINS];
+
+/**
+ * Validates that the provided domain and chainId combination is correct
+ * TypeScript will enforce that only valid domain/chainId pairs are accepted
+ */
+export function validateDomainChain(config: DomainChainConfig): boolean {
+  const domainInfo = Object.values(DOMAINS).find(
+    (info) => info.domain === config.domain,
+  );
+  return !!domainInfo && domainInfo.chainId === config.chainId;
+}
+
+/**
+ * Generic validation result interface
+ */
+export interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+/**
+ * TACo Configuration Validator
+ *
+ * Validates TACo client configurations, domains, and provider compatibility.
+ * Provides both fast and full validation methods for TACo operations.
+ */
+export class TacoConfigValidator {
+  /**
+   * Get all supported TACo domain names
+   * @returns {DomainName[]} Array of supported TACo domain names ('lynx', 'tapir', 'mainnet')
+   */
+  static getSupportedDomains(): DomainName[] {
+    return Object.values(DOMAINS).map((domain) => domain.domain);
+  }
+
+  /**
+   * Check if domain is valid
+   * @param domain - TACo domain name to check ('lynx', 'tapir', 'mainnet')
+   * @returns {boolean} True if domain exists
+   */
+  static isValidDomain(domain: DomainName): boolean {
+    return !!domain && this.getSupportedDomains().includes(domain);
+  }
+
+  /**
+   * Get expected chain ID for domain from DOMAINS configuration
+   * @param domain - Domain name to look up
+   * @returns {number | undefined} Chain ID for the domain, undefined if not found
+   * @private
+   */
+  private static getExpectedChainId(domain: string): number | undefined {
+    const domainEntry = Object.values(DOMAINS).find(
+      (domainConfig) => domainConfig.domain === domain,
+    );
+    return domainEntry?.chainId;
+  }
+
+  /**
+   * Validate ritual ID (basic validation - positive number only)
+   * @param domain - Domain name (unused but kept for API compatibility)
+   * @param ritualId - Ritual ID to validate
+   * @returns {boolean} True if valid (positive number)
+   */
+  static isValidRitualId(domain: DomainName, ritualId: number): boolean {
+    return typeof ritualId === 'number' && ritualId > 0;
```

Comment:
Isn't the typeof already satisfied since ritualId is a parameter of type number?